server/cmwell-data-tools/src/main/scala/cmwell/tools/data/sparql/SparqlProcessor.scala

```diff
@@ -173,8 +174,23 @@ class SparqlProcessor[T](baseUrl: String,
     type Paths = Seq[ByteString]
     type StartTime = Long
 
-    def validateResponseBody(body: ByteString) = {
-      !(body containsSlice "Could not process request")
+    def validateResponse(body: ByteString, headers: Seq[HttpHeader]) : Try[Unit] = {
+      def validateBody(body: ByteString) = {
+        if (!(body containsSlice "Could not process request")) None
+        else Some("Response body was not valid")
+      }
+
+      def validateHeaders(headers: Seq[HttpHeader])  = {
+        if(!(headers.contains("X-CM-WELL-SG-RS"))) None
+        else Some("Error reported in trailer headers")
```

Comment:
please expose the actual header (with the failed error numbers). Talked to @bryaakov and he said it will help analyzing errors.

Reply:
Agree. Implemented